charts/external-dns/templates/clusterrole.yaml

```diff
@@ -1,12 +1,12 @@
 {{- if .Values.rbac.create -}}
 apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRole
+kind: {{ .Values.namespaced | ternary "Role" "ClusterRole" }}
 metadata:
   name: {{ template "external-dns.fullname" . }}
   labels:
     {{- include "external-dns.labels" . | nindent 4 }}
 rules:
-{{- if or (has "node" .Values.sources) (has "pod" .Values.sources) (has "service" .Values.sources) (has "contour-httpproxy" .Values.sources) (has "gloo-proxy" .Values.sources) (has "openshift-route" .Values.sources) (has "skipper-routegroup" .Values.sources) }}
+{{- if and (not .Values.namespaced) (or (has "node" .Values.sources) (has "pod" .Values.sources) (has "service" .Values.sources) (has "contour-httpproxy" .Values.sources) (has "gloo-proxy" .Values.sources) (has "openshift-route" .Values.sources) (has "skipper-routegroup" .Values.sources)) }}
```

Comment:
I think this needs documenting if some sources don't work in namespaced mode. Would I be right in thinking that CRDs also wouldn't work an that they might also need to be gated (maybe a wrapping if block for not namespaced)?

Reply:
CRDs are working in namespace mode, since the CRs are namespaced. Only the installation of CRDs is not possible.